docs/DocGenerator/README.md

```diff
@@ -0,0 +1,205 @@
+
+- **Output path**: The path where all the parsed documentation blocks will be dumped.
+
+- **Include paths**: A list with all the paths from where the tool will look to extract and parse the documentation blocks
+
+- **Include regex**: A list of regexes to identify which files must be considered a test file.
+
+- **Group files**: A list of regexes to identify which files must be considered a group file.
+
+- **Function regex**: A list of regexes to identify which methods inside a test file must be considered a test method.
+
+- **Ignore paths**: A list with the paths that should be bypassed during the parsing scan.
+
+- **Output fields**: Lists with the expected fields in the documentation. There is a list for the modules fields and
+another for test fields. In turn, each of them is divided into Mandatory and Optional fields.
+During a sanity check, it will be checked that every Mandatory field is present in the respective documentation block.
+During the parsing, every field not present in the Mandatory or the Optional fields will be ignored.
+
+- **Test cases field**: Key to identify the test cases of each test. If this field is missing in the documentation block
+of a test, the tool will execute a dry run of pytest to automatically find all the test cases of the test and append
+this information with the key specified. If this option is missing, or if the key is found in the test documentation
+block, this action is avoided.
+
+## Usage
+
+### Parsing
+    python3 DocGenerator.py
+
+Without using any flag, the tool will load the configuration file and run a complete parse of the paths in the configuration
+to dump the content into the output folder.
+
+### Sanity Check
+    python3 DocGenerator.py -s
+
+Using **-s**, the tool will run a sanity check of the content in the output folder.
+
+It will check the coverage of the already parsed files in the **Output path** comparing it with the tests found in
+**Project path**.
+Also, it will validate that the output files have every Mandatory field.
+It will also list and report every tag found.
+
+### Debug
+    python3 DocGenerator.py -d
+
+Using **-d**, the tool will run in DEBUG mode, logging extra information in the log file.
+
+### Version
+    python3 DocGenerator.py -v
+
+Using **-v**, the tool will print its current version.
+
+### Config Test
+    python3 DocGenerator.py -t
+Using **-t** option, the tool will load the config file to test the correct content.
+
+## Index output data
+    python3 DocGenerator.py -i <INDEX_NAME>
+Using **-i** option, the tool will load the content of the output folder to index it into Elastic. The name of the index
+must be provided as a parameter
+
+## Local launch output data
+    python3 DocGenerator.py -l <INDEX_NAME>
+Using **-l** option, the tool will index the content of the output folder and lunch SearchUI app locally. The name of the index
+must be provided as a parameter
```

Comment:
I would like something like: "... the tool indexes the content of each file output as a document into ElasticSearch and launch the application. The name of the index must be provided as a parameter"